provider/src/main/kotlin/org/gradle/kotlin/dsl/support/EmbeddedKotlinProvider.kt

```diff
@@ -34,7 +34,7 @@ import java.util.*
 
 
 private
-const val embeddedRepositoryCacheKeyVersion = 1
+const val embeddedRepositoryCacheKeyVersion = 2
```

Comment:
This is a cache busting mechanism right?
I think I probably understand what this is, but can you add a quick comment above it explaining what it is and why it's necessary.

Also, why is the bump necessary?

Reply:
See https://github.com/gradle/kotlin-dsl/pull/693#issuecomment-360105473